sourcegraph/sourcegraph
# svelte: Inject window.context when serving the Svelte index.html page

Closes #61307 

This commit extends the server to parse SvelteKit's index.html file as a template and inject `window.context`.
The SvelteKit app doesn't use window.context yet but this change fixes an issue when running `sg start web-standalone` against S2 and S2 is configured to serve the SvelteKit app.

Couple of notes:

- I originally wanted to use a single `app.html` file for both "integrated" and standalone development. Having two separate files is not ideal, but was the easier way.
- I mostly copied the logic for generating and caching the template from the code that serves the esbuild bundle. There didn't seem to be an obvious way to make that reusable.


## Test plan

Manual testing by running

```
sg start enterprise
pnpm dev:enterprise
```

and inspecting the HTML returned by `https://sourcegraph.test:3443`

![2024-03-20_20-49](https://github.com/sourcegraph/sourcegraph/assets/179026/29c3bab7-bdb5-4574-9203-416e7760d114)

I couldn't verify that it indeed works with `sg run web-standalone` because I can't run an instance and the standalone web app, but I looked at our esbuild devserver setup and verified that we indeed extract `window.context` from the origin server response.
